src/providers/data_provider_be.c

```diff
@@ -249,8 +249,9 @@ static errno_t be_check_online_request(struct be_ctx *be_ctx)
 static void be_check_online_done(struct tevent_req *req)
 {
     struct be_ctx *be_ctx;
-    struct dp_reply_std reply;
+    struct dp_reply_std *reply;
     errno_t ret;
+    static int last_dp_state = -1;
```

Comment:
I'm abvivalent about a static variable, because I don't think functions should have state, but I guess in the end it is a bit cleaner than putting another field in the be context as long as this variable is only used for logging, but could you please add a comment that this is only about logging so that nobody tries to optimize the code in the future?

Reply:
Well, I can either add this state to the be_context or a comment to the static var.
Just let me know what's your preference.